src/firmware/types.rs

```diff
@@ -202,3 +210,512 @@ pub struct SnpPlatformStatus {
+    #[cfg_attr(feature = "openssl", serde(with = "BigArray"))]
+    pub report_data: [u8; 64],
+    /// The measurement calculated at launch.
+    #[cfg_attr(feature = "openssl", serde(with = "BigArray"))]
+    pub measurement: [u8; 48],
+    /// Data provided by the hypervisor at launch.
+    pub host_data: [u8; 32],
+    /// SHA-384 digest of the ID public key that signed the ID block provided
+    /// in SNP_LANUNCH_FINISH.
+    #[cfg_attr(feature = "openssl", serde(with = "BigArray"))]
+    pub id_key_digest: [u8; 48],
+    /// SHA-384 digest of the Author public key that certified the ID key,
+    /// if provided in SNP_LAUNCH_FINSIH. Zeroes if AUTHOR_KEY_EN is 1.
+    #[cfg_attr(feature = "openssl", serde(with = "BigArray"))]
+    pub author_key_digest: [u8; 48],
+    /// Report ID of this guest.
+    pub report_id: [u8; 32],
+    /// Report ID of this guest's migration agent (if applicable).
+    pub report_id_ma: [u8; 32],
+    /// Reported TCB version used to derive the VCEK that signed this report.
+    pub reported_tcb: SNPTcbVersion,
+    _reserved_1: [u8; 24],
+    /// If MaskChipId is set to 0, Identifier unique to the chip.
+    /// Otherwise set to 0h.
+    #[cfg_attr(feature = "openssl", serde(with = "BigArray"))]
+    pub chip_id: [u8; 64],
+    /// CommittedTCB
+    pub committed_tcb: SNPTcbVersion,
+    /// The build number of CurrentVersion
+    pub current_build: u8,
+    /// The minor number of CurrentVersion
+    pub current_minor: u8,
+    /// The major number of CurrentVersion
+    pub current_major: u8,
+    _reserved_2: u8,
+    /// The build number of CommittedVersion
+    pub committed_build: u8,
+    /// The minor number of CommittedVersion
+    pub committed_minor: u8,
+    /// The major number of CommittedVersion
+    pub committed_major: u8,
+    _reserved_3: u8,
+    /// The CurrentTcb at the time the guest was launched or imported.
+    pub launch_tcb: SNPTcbVersion,
+    #[cfg_attr(feature = "openssl", serde(with = "BigArray"))]
+    _reserved_4: [u8; 168],
+    /// Signature of bytes 0 to 0x29F inclusive of this report.
+    /// The format of the signature is found within [`Signature`].
+    pub signature: Signature,
+}
+
+/// The response from the PSP containing the generated attestation report.
+/// According to the kernel, this should match [`self::guest::ReportRsp`].
+/// The Report is padded to exactly 4000 Bytes to make sure the page size
+/// matches.
+///
+///
+/// ```txt
+///     96 Bytes (*Message Header)
+/// + 4000 Bytes (*Encrypted Message)
+/// ------------
+///   4096 Bytes (4k Memory Page Alignment)
+/// ```
+/// <sup>*[Message Header - 8.26 SNP_GUEST_REQUEST - Table 97](<https://www.amd.com/system/files/TechDocs/56860.pdf#page=113>)</sup>
+///
+/// <sup>*[Encrypted Message - sev-guest.h](<https://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git/tree/include/uapi/linux/sev-guest.h>)</sup>
+#[repr(C)]
+pub struct MsgReportRsp {
+    /// The status of key derivation operation.
```

Comment:
```suggestion
    /// The status of attestation report generation.
```

(yes, it's a mistake also in the SNP spec table 23.)

Reply:
Fixing